content/docs/userspace/full-stack/1-intro.md

```diff
@@ -0,0 +1,84 @@
++++
+title = "1. Introduction"
+weight = 1
+template = "doc.html"
++++
+
+This series walks through the writing of a full Gall agent, and then the process
+of integrating it with a React front-end. This series follows on from the
+previous [Gall Guide](/docs/userspace/gall-guide/intro). If you haven't
+completed that, or otherwise aren't familiar with the basics of writing Gall
+agents, it's strongly recommended to work through that guide first.
+
+The app we'll be looking at is a simple journal with an agent called `%journal`.
+In the browser, users will be able to add plain text journal entries organized
+by date. Entries may be scrolled through in ascending date order, with more
+entries loaded each time the bottom of the list is reached. Old entries will be
+able to be edited and deleted, and users will be able to search through entries
+by specifying a date range.
+
+The `Journal` app we'll be looking at can be installed on a live ship from
+`~pocwet/journal`, and its source code is available [here](https://github.com/urbit/docs-examples/tree/main/journal-app).
+
+![journal ui screenshot](https://media.urbit.org/docs/userspace/full-stack-guide/entries.png)
+
+This walkthrough does not contain exercises, nor does it completely cover every
```

Comment:
May be worth including a few prompts tho:  "how would you include formatting?" type of thought questions.

Reply:
I can't think of a way to do this that's not just unnaturally shoehorning in random questions that don't actually help grok the concepts being discussed. It's a good idea in highly constrained guides where you're trying to hammer very particular points but there are essentially infinite ways to vary this program